### DISCUSSION TITLE
PSD to dds image

### DISCUSSION THREAD

**Discussion by wandersonareis** [ImageMagick/ImageMagick#2748] [Help]
Converting this PSD [0_14_256_20_0_PSD.zip](https://github.com/ImageMagick/ImageMagick/files/5411175/0_14_256_20_0_PSD.zip)
Image converted by photoshop Nvidia plugin [0_14_256_20_0_DDS_nvidia.zip](https://github.com/ImageMagick/ImageMagick/files/5411187/0_14_256_20_0_DDS_nvidia.zip)
Image converted by ImageMagick [0_14_256_20_0_ImageMagick.zip](https://github.com/ImageMagick/ImageMagick/files/5411196/0_14_256_20_0_ImageMagick.zip)

This: convert.exe -define dds:mipmaps=0 -flip

The image made by the nvidia plugin is the same as the original .dds image in number of colors.
When I use to convert the generated image it has less unique colors.
390 unique colors in the plugin image versus 102 unique colors in the converter.
Is there anything to preserve the number of unique colors?

![2020-10-20 (1)](https://user-images.githubusercontent.com/9261306/96637669-61153780-12f5-11eb-93a0-f50796f6c2dd.png)

**Comment by snibgo** [comment]
What version of IM? On what platform?

> This: convert.exe -define dds:mipmaps=0 -flip

That isn't a valid IM command. Please show the command you used.

**Comment by wandersonareis** [comment]
This full command: convert.exe -define dds:mipmaps=0 C:\Users\wande\Music\0_14_256_20_0.psd C:\Users\wande\Music\0_14_256_20_0.dds

ImageMagick-7.0.10-34-portable-Q16-HDRI-x64

Using windows 10

**Reply by fmw42** [reply]
Try reading the input psd file before -define. Since you are using IM 7, use magick not convert.

**Reply by wandersonareis** [reply]
It's the same.
magick.exe -format dds -define dds:compression=dxt5 C:\Users\wande\Music\0_14_256_20_0.psd C:\Users\wande\Music\0_14_256_20_0.dds

**Comment by fmw42** [comment]
Read the input before the define. Try DDS: prefix to the output. Also read only the flattened PSD layer [0]

`magick.exe C:\Users\wande\Music\0_14_256_20_0.psd[0] -define dds:compression=dxt5 C:\Users\wande\Music\DDS:0_14_256_20_0.dds`

If that fails, try changing directory to the input directory and then just use the file names

`magick.exe 0_14_256_20_0.psd[0] -define dds:compression=dxt5 DDS:0_14_256_20_0.dds`

**Reply by wandersonareis** [reply]
102 colors.
magick.exe psd\0_14_256_20_0.psd[0] -define dds:compression=dxt5 DDS\0_14_256_20_0.dds

PSD edited with one layer. Layer rasterized. It's the same.

I have less colors in other images as well.
In color images the color reduction is smaller and less apparent.
This image is visibly degraded by the reduction of colors.

**Comment by snibgo** [comment]
What tool are you using to count colours?

```
magick 0_14_256_20_0.psd[0] -unique-colors info:

0_14_256_20_0.psd[0] PSD 201x1 201x1+0+0 8-bit sRGB 101049B 0.000u 0:00.007
```
So IM finds 201 unique colours in the [0] image.

We write a DDS file, and count the colours in that:
```
magick 0_14_256_20_0.psd[0] -define dds:compression=dxt5 x.dds

magick x.dds -unique-colors info:

x.dds DDS 536x1 536x1+0+0 8-bit sRGB 11152B 0.016u 0:00.006
```
So the number of colours has increased to 536. I suppose this is a result of the compression.

**Reply by wandersonareis** [reply]
I'm using xnview to count colors.
Use online converter like [Convertio](https://convertio.co/pt/psd-dds/) is the same result.
Xnconverter same result.
This color degradation is visible.
![image](https://user-images.githubusercontent.com/9261306/96817606-0ca5b100-13f6-11eb-8276-e55d931f38b0.png)
